presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -908,6 +910,9 @@ protected RowExpression visitLikePredicate(LikePredicate node, Context context)
 
             if (node.getEscape().isPresent()) {
                 RowExpression escape = process(node.getEscape().get(), context);
+                if (coordinatorSidecarEnabled) {
```

Comment:
Instead of switching on `coordinatorSidecarEnabled`, can we introduce a method `functionResolution.supportsLikePattern`, which just checks if the lake pattern function is present.  If so, use it, otherwise, use a varchar version.